src/details/ArborX_DetailsDistributedTreeNearestHelpers.hpp

```diff
@@ -93,6 +93,76 @@ struct WithinDistanceFromPredicates
   Predicates predicates;
   Distances distances;
 };
+
+namespace Dispatch
+{
+
+template <typename Tag, typename Geometry>
+struct approx_expand_by_radius;
+
+template <typename Point>
+struct approx_expand_by_radius<PointTag, Point>
+{
+  template <typename Float>
+  KOKKOS_FUNCTION static auto apply(Point const &point, Float r)
+  {
+    constexpr int DIM = GeometryTraits::dimension_v<Box>;
+    using Coordinate = GeometryTraits::coordinate_type_t<Point>;
+    auto const &hyper_point = reinterpret_cast<
+        ExperimentalHyperGeometry::Point<DIM, Coordinate> const &>(point);
```

Comment:
Is there any good reason not just to copy the data (possibly via `bit_cast`)? I'm pretty uncomfortable with using `reinterpret_cast` here. Ultimately the point is copied into `Sphere` anyway and I would expect that the compiler could avoid copying twice.

Reply:
Done.